src/test/resources/tck/openapi3/discriminator.json

```diff
@@ -74,5 +74,174 @@
         "valid": false
       }
     ]
+  },
+  {
+    "description": "discriminator with oneOf validation with ref",
+    "schema": {
+      "oneOf": [
+        {
+          "$ref": "#/definitions/A"
+        },
+        {
+          "$ref": "#/definitions/B"
+        }
+      ],
+      "discriminator": {
+        "propertyName": "id"
+      },
+      "definitions": {
+        "A": {
+          "type": "object",
+          "properties": {
+            "id": {
+              "type": "string"
+            },
+            "value_a": {
+              "type": "string"
+            }
+          },
+          "required": ["id", "value_a"]
+        },
+        "B": {
+          "type": "object",
+          "properties": {
+            "id": {
+              "type": "string"
+            },
+            "value_b": {
+              "type": "string"
+            }
+          },
+          "required": ["id"]
+        }                
+      }
+    },
+    "tests": [
+      {
+        "description": "should be valid with id A and property value_a",
+        "data": {
+          "id": "A",
+          "value_a": "a"
+        },
+        "valid": true
+      },
+      {
+        "description": "should be invalid with id A and property value_b",
+        "data": {
+          "id": "A",
+          "value_b": "b"
+        },
+        "valid": false
```

Comment:
Why this is invalid? This should be valid because there isn't any `additionalProperties: false` in the A schema, so `value_b` should be catched as additional property

Reply:
Same question for the same test case below

Reply:
It is because `"required": ["id", "value_a"]` in the A schema